### PR TITLE
[AIRFLOW-4296][AIP-3-subtask-18] Remove py2 in ci process

### PR DESCRIPTION
**Pull request by zhongjiajie**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4296

### Description

My PR https://github.com/apache/airflow/pull/5020 CI failed, and the sample log https://api.travis-ci.org/v3/job/519213992/log.txt

```sh
Traceback (most recent call last):
  File "/usr/local/bin/airflow", line 21, in <module>
    from airflow import configuration
  File "/usr/local/lib/python2.7/dist-packages/airflow/__init__.py", line 36, in <module>
    from airflow import settings, configuration as conf
  File "/usr/local/lib/python2.7/dist-packages/airflow/settings.py", line 32, in <module>
    from airflow.utils.sqlalchemy import setup_event_handlers
  File "/usr/local/lib/python2.7/dist-packages/airflow/utils/sqlalchemy.py", line 22, in <module>
    import json
  File "/usr/local/lib/python2.7/dist-packages/airflow/utils/json.py", line 27, in <module>
    class AirflowJsonEncoder(json.JSONEncoder):
AttributeError: 'module' object has no attribute 'JSONEncoder'
```

And I found out we ci process still have py2, So remove `from __future__ import absolute_import` will cause error. This PR are try to fix this.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by zhongjiajie** [comment]
@Fokko @ashb @XD-DENG  PTAL.

**Comment by kaxil** [comment]
This is already done and merged in master:

https://github.com/apache/airflow/commit/3020d9733cb189f091489a62f58f0f586dc8d4a9

**Comment by zhongjiajie** [comment]
@kaxil I rebase on master but PR-5019 and PR-5020 still failed.

**Comment by Fokko** [comment]
Restarted the CI

**Comment by zhongjiajie** [comment]
Still failed, I try to fix it.

File: scripts/ci/kubernetes/docker/airflow-test-env-init.sh
Comment (jmcarp) [inline scripts/ci/kubernetes/docker/airflow-test-env-init.sh]: It looks like this path should use `python3.5` instead of `python3.6`. Xenial LTS installs `python3.5` unless we add a new PPA.

Comment (zhongjiajie) [reply scripts/ci/kubernetes/docker/airflow-test-env-init.sh]: Thanks for catching. But I change base docker images to python:3.6-slim because we have a jira ticket in https://issues.apache.org/jira/browse/AIRFLOW-4210. And I think we should change our ci k8s as well.

**Comment by zhongjiajie** [comment]
Add gcc because of failed in https://travis-ci.org/apache/airflow/jobs/520123725

**Comment by zhongjiajie** [comment]
CI green. But before merge I'd like to test if this change work in my personal repo https://github.com/zhongjiajie/airflow/tree/test_if_ci_work

Travis CI log is here https://travis-ci.com/zhongjiajie/airflow/builds/108322089

I use this PR and cherry-pick https://github.com/apache/airflow/pull/5020. submit https://github.com/zhongjiajie/airflow/tree/test_if_ci_work

**Comment by zhongjiajie** [comment]
Travis CI log is here https://travis-ci.com/zhongjiajie/airflow/builds/108322089

**Comment by Fokko** [comment]
@zhongjiajie Still a sad CI :'( Can you rerun the job?

**Comment by zhongjiajie** [comment]
Seem failed with un-related test https://travis-ci.com/zhongjiajie/airflow/jobs/192913602
But I will try again to ensure.

**Comment by zhongjiajie** [comment]
https://travis-ci.com/zhongjiajie/airflow/builds/108333920

K8S ci pass, but postgre failed and I think the failed is un-related

```sh
======================================================================
46) ERROR: test_execution_limited_parallelism (tests.executors.test_local_executor.LocalExecutorTest)
----------------------------------------------------------------------
   Traceback (most recent call last):
    tests/executors/test_local_executor.py line 75 in test_execution_limited_parallelism
      self.execution_parallelism(parallelism=test_parallelism)
    tests/executors/test_local_executor.py line 61 in execution_parallelism
      self.assertTrue(executor.event_buffer['fail'], State.FAILED)
   KeyError: 'fail'
```

And this postgre job sucess in https://travis-ci.com/zhongjiajie/airflow/builds/108322089

Should I restart the CI in my repo or just accept/review this PR
WDYT @Fokko

**Comment by milton0825** [comment]
@zhongjiajie can you restart the tests in your CI? Probably just a flaky test.

**Comment by zhongjiajie** [comment]
@milton0825 I think is flaky test, but in job https://travis-ci.com/zhongjiajie/airflow/builds/108333920 we could found out K8S test job pass(we just want to prove K8S test pass). So I wonder if we could accept this PR

**Comment by zhongjiajie** [comment]
https://travis-ci.org/zhongjiajie/airflow/builds/520548126

FYI @Fokko @milton0825 , CI green in my own repo https://github.com/zhongjiajie/airflow/tree/test_if_ci_work

**Comment by milton0825** [comment]
lgtm

**Comment by jmcarp** [comment]
Anything left to do on this? It looks good to me.

**Comment by mik-laj** [comment]
PTAL @potiuk

**Comment by milton0825** [comment]
Yeah let's ship this as it blocks the PY2 deprecation.

**Review by potiuk — APPROVED** [review]
Looks good. Hopefully I got to fix the last issue with multi-staging docker soon and we get rid of it altogether ;)

**Comment by zhongjiajie** [comment]
Ping @Fokko @ashb  @kaxil @XD-DENG PTAL.

**Comment by zhongjiajie** [comment]
@potiuk Do you have the rights to merge PR now. I check my dev-email and notice you become committer. BTW Congratulations :+1:

**Comment by zhongjiajie** [comment]
@potiuk Thanks for merging.

**Comment by zhongjiajie** [comment]
@potiuk Notice that you don't change jira ticket status as **RESOLVED**, maybe you should take a look as https://github.com/apache/airflow/pull/4966#issuecomment-476243906. FYI

**Comment by potiuk** [comment]
Yeah..I realised I have no necessary rights in JIRA yet. I am waiting for infra team to grant me the right. But I keep the list of tickets I merged so no worries ! 

Happy Easter :)

**Comment by Fokko** [comment]
Thanks @potiuk for picking this up. Happy Easter as well! 👍

**Pull request merged**
